docs/security_self_assessment.md

```diff
@@ -0,0 +1,1588 @@
+
+## Controls
+
+### 1 Master Node Security Configuration
+
+#### 1.1 Master Node Configuration Files
+
+*1.1.1* Ensure that the API server pod specification file permissions are set to `644` or more restrictive (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for the API server. All configuration is passed in as arguments at container run time.
+
+*1.1.2* Ensure that the API server pod specification file ownership is set to `root:root` (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for the API server. All configuration is passed in as arguments at container run time.
+
+*1.1.3* Ensure that the controller manager pod specification file permissions are set to `644` or more restrictive (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for the controller manager. All configuration is passed in as arguments at container run time.
+
+*1.1.4* Ensure that the controller manager pod specification file ownership is set to `root:root` (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for the controller manager. All configuration is passed in as arguments at container run time.
+
+*1.1.5* Ensure that the scheduler pod specification file permissions are set to `644` or more restrictive (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for the scheduler. All configuration is passed in as arguments at container run time.
+
+*1.1.6* Ensure that the scheduler pod specification file ownership is set to `root:root` (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for the scheduler. All configuration is passed in as arguments at container run time.
+
+*1.1.7* Ensure that the etcd pod specification file permissions are set to `644` or more restrictive (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for etcd. All configuration is passed in as arguments at container run time.
+
+*1.1.8* Ensure that the etcd pod specification file ownership is set to `root:root` (Scored)
+
+Result: Not Applicable
+
+Remediation: RKE2 doesn’t require or maintain a configuration file for etcd. All configuration is passed in as arguments at container run time.
+
+*1.1.11* Ensure that the etcd data directory permissions are set to `700` or more restrictive (Scored)
+
+Result: Not Applicable
+
+Remediation: On the etcd server node, get the etcd data directory, passed as an argument `--data-dir`, from the below command:
```

Comment:
This remediation won't work for rke2. Its for rke1. our data dir is not at /var/lib/etcd. The 1.1.11.sh script assumes docker, which is not on the host.

Reply:
I have additional related comments in the security-scan PR around this control